lib/IR/AsmWriter.cpp

```diff
@@ -3366,6 +3366,30 @@ void MDNode::printAsBody(raw_ostream &OS, ModuleSlotTracker &MST, const Module *
     TypePrinter.incorporateTypes(*M);
   WriteMDNodeBodyInternal(OS, this, &TypePrinter, MST.getMachine(), M);
 }
+
+template<typename T>
+static void dumpToFileImpl(const T *obj, const char *path) {
+  std::string s;
+  llvm::raw_string_ostream os(s);
+  os << *obj;
+  os.flush();
+
+  size_t bytesWritten = 0;
+  FILE *f = fopen(path, "wb");
+  if (f) {
+    bytesWritten = fwrite(s.c_str(), 1, s.size(), f);
+    fclose(f);
+  }
+  fprintf(stderr, "%u bytes written to %s\n", (unsigned)bytesWritten, path);
```

Comment:
Does this circumvent the per-thread file system, or do we override the fprintf implementation?  I don't think we want to circumvent the file system stuff we have in place.

Reply:
Oh yeah, and same question for above `fopen` etc...

Reply:
This is just for calling in the debugger, so it's circumventing the per thread file system deliberately.

Reply:
That's a good idea. Done.